File: kakao/2018/kakao2018_5.cpp
#include <bits/stdc++.h>
using namespace std;
int a,b,c[676],d[676],i;

int solution(string str1,string str2)
{
	for(i=1;i<str1.size();i++)
		if(isalpha(str1[i-1])&&isalpha(str1[i]))
			c[((str1[i-1]&31)-1)*26+(str1[i]&31)-1]++;
	for(i=1;i<str2.size();i++)
		if(isalpha(str2[i-1])&&isalpha(str2[i]))
			d[((str2[i-1]&31)-1)*26+(str2[i]&31)-1]++;
	for(i=0;i<676;i++)
		a+=min(c[i],d[i]),b+=max(c[i],d[i]);
	return b?a*65536/b:65536;
}

File: kakao/2018/kakao2018_5.py
def solution(str1, str2):
    a=b=0;c=[0]*676;d=[0]*676
    for i in range(len(str1)-1):
        if str1[i].isalpha() and str1[i+1].isalpha():
            c[((ord(str1[i])&31)-1)*26+(ord(str1[i+1])&31)-1]+=1
    for i in range(len(str2)-1):
        if str2[i].isalpha() and str2[i+1].isalpha():
            d[((ord(str2[i])&31)-1)*26+(ord(str2[i+1])&31)-1]+=1
    for i in range(676):a+=min(c[i],d[i]);b+=max(c[i],d[i])
    return a*65536//b if b else 65536